oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/geometry/Geo.java

```diff
@@ -300,6 +336,195 @@ private static double ringArea(LinearRing ring) {
+  // ======================
+
+  /**
+   * Calculates the "Polsby–Popper test" score of a polygonal geometry.
+   *
+   * <p>
+   * If the shape is not polygonal, zero is returned. If a shape constitues of multiple parts (a
+   * MultiPolygon geometry), the result is the weighted sum of the individual parts' scores.
+   * See <a href="https://en.wikipedia.org/wiki/Polsby%E2%80%93Popper_test">wikipedia</a> for info.
+   * </p>
+   *
+   * @param geom the geometry for which to calculate the PP score of.
+   * @return the compaceness measure of the input shape. A score of 1 indicates maximum compactness
+   *         (i.e. a circle)
+   */
+  public static double compactness(Geometry geom) {
+    if (!(geom instanceof Polygonal)) {
+      return 0;
+    }
+    var boundaryLength = Geo.lengthOf(geom.getBoundary());
+    if (boundaryLength == 0) {
+      return 0;
+    }
+    return 4 * Math.PI * Geo.areaOf(geom) / (boundaryLength * boundaryLength);
+  }
+
+  // ======================
+  // = angle calculations =
+  // ======================
+
+  public static double bearing(Coordinate from, Coordinate to) {
+    return bearingRadians(from, to) * 180 / Math.PI;
+  }
+
+  public static double bearingRadians(Coordinate from, Coordinate to) {
+    var x1 = from.x * Math.PI / 180;
+    var x2 = to.x * Math.PI / 180;
+    var y1 = from.y * Math.PI / 180;
+    var y2 = to.y * Math.PI / 180;
+    var y = Math.sin(x2 - x1) * Math.cos(y2);
+    var x = Math.cos(y1) * Math.sin(y2)
+        - Math.sin(y1) * Math.cos(y2) * Math.cos(x2 - x1);
+    return (Math.atan2(y, x) + 2 * Math.PI) % (2 * Math.PI);
+  }
+
+  /**
+   * Returns a measure for the rectilinearity (or squareness) of a geometry.
+   *
+   * <p>Adapted from "A Rectilinearity Measurement for Polygons" by Joviša Žunić and Paul L. Rosin:
+   * DOI:10.1007/3-540-47967-8_50
+   * https://link.springer.com/chapter/10.1007%2F3-540-47967-8_50
+   * https://www.researchgate.net/publication/221304067_A_Rectilinearity_Measurement_for_Polygons
+   *
+   * Žunić J., Rosin P.L. (2002) A Rectilinearity Measurement for Polygons. In: Heyden A., Sparr G.,
+   * Nielsen M., Johansen P. (eds) Computer Vision — ECCV 2002. ECCV 2002. Lecture Notes in Computer
+   * Science, vol 2351. Springer, Berlin, Heidelberg. https://doi.org/10.1007/3-540-47967-8_50
+   * </p>
+   *
+   * <p>Adjusted to work directly on geographic coordinates. Implementation works on a few
+   * assumptions: input geometries are "small"; spherical globe approximation.</p>
+   *
+   * @param geom a Polygon, MultiPolygon and LineString for which the squareness is to be evaluated.
+   * @return returns the rectilinearity value of the input geometry, or zero if the geometry type
+   *         isn't supported
+   */
+  public static double rectilinearity(Geometry geom) {
+    LineString[] lines;
+    if (geom instanceof Polygon) {
+      var poly = (Polygon) geom;
+      var rings = new ArrayList<LinearRing>(poly.getNumInteriorRing() + 1);
```

Comment:
Could be done for consistency with function below.
```suggestion
      var rings = new ArrayList<LinearRing>();
```